src/nrnpython/nrnpy_hoc.cpp

```diff
@@ -3106,9 +3127,18 @@ PyObject* nrnpy_hoc() {
     if (PyType_Ready(hocobject_type) < 0)
         goto fail;
     Py_INCREF(hocobject_type);
+    //printf("defining vectorobject_type\n");
+    vectorobject_type = (PyTypeObject*) PyType_FromSpecWithBases(&nrnpy_HocObjectType_spec, NULL);//(PyObject*) hocobject_type);
+    if (PyType_Ready(vectorobject_type) < 0)
+        goto fail;
+    Py_INCREF(vectorobject_type);
+    //printf("done\n");
     // printf("AddObject HocObject\n");
     PyModule_AddObject(m, "HocObject", (PyObject*) hocobject_type);
-
+    //printf("registering Vector\n");
+    PyModule_AddObject(m, "Vector", (PyObject*) vectorobject_type);
```

Comment:
You need to set `vectorobject_type.tp_base = hocobject_type` to indicate it as a subclass.

On my phone so may have missed this happening somewhere